packages/compass-import-export/src/import/analyze-csv-fields.ts

```diff
@@ -141,77 +137,81 @@ function pickFieldType(field: CSVField): CSVParsableFieldType {
   return field.detected;
 }
 
-export function analyzeCSVFields({
+export async function analyzeCSVFields({
   input,
   delimiter,
   abortSignal,
   progressCallback,
   ignoreEmptyStrings,
 }: AnalyzeCSVFieldsOptions): Promise<AnalyzeCSVFieldsResult> {
-  return new Promise(function (resolve, reject) {
-    const byteCounter = new ByteCounter();
-
-    const result: AnalyzeCSVFieldsResult = {
-      totalRows: 0,
-      fields: {},
-      aborted: false,
-    };
-
-    let aborted = false;
-    let headerFields: string[];
-
-    const validator = new Utf8Validator();
-
-    validator.once('error', function (err: any) {
-      reject(err);
-    });
-
-    input = input.pipe(validator).pipe(byteCounter).pipe(stripBomStream());
-
-    Papa.parse(input, {
-      delimiter,
-      header: true,
-      step: function (results: Papa.ParseStepResult<PapaRowData>, parser) {
-        if (abortSignal?.aborted && !aborted) {
-          aborted = true;
-          result.aborted = true;
-          parser.abort();
```

Comment:
Oh and like listCSVFields() this also didn't properly destroy the input stream whereas `await pipeline()` wit the signal passed in does.